initdb.py

```
@@ -59,75 +90,85 @@ def create_database(cursor, connection):
-connection_cursor = mysql_connection.cursor()
-# Set autocommit to false for batch
-mysql_connection.autocommit = False
-# Create database
-create_database(connection_cursor, mysql_connection)
-# Create tables
-create_tables(connection_cursor, mysql_connection)
-# Close cursor
-connection_cursor.close()
-# Close connection
-mysql_connection.close()
+
+def insert_data(cursor):
+    complete = 1
+    for name, data in DATASETS.items():
+        next(data)
+        total = len(DATASETS)
+        print("({}/{})Inserting data into {}".format(complete, total, name))
+        complete += 1
+        for row in data:
+            cursor.execute(SQL_INSERT[name], row)
+
+
+def main(username='', password=''):
+    if username == "" or password == "":
+        # Get username and password for desired account
+        username = input("Username (root or other account): ")
+        password = input("Password: ")
+
+    # Connect to the MySQL server with user credentials
+    # Will exit if MySQL Server is not started
+    try:
+        mysql_connection = mysql.connector.connect(user=username,
+                                                   password=password)
+    except mysql.connector.Error as err:
+        raise Exception('There was a problem connecting to the server:\n{}'.format(err))
+
+    # Get cursor from server connection
+    mysql_cursor = mysql_connection.cursor()
+    # Set autocommit to false for batch
+    mysql_connection.autocommit = False
+
+    # Start a transaction
+    mysql_connection.start_transaction()
+
+    # Create database
+    create_database(mysql_cursor)
+    # Create tables
+    create_tables(mysql_cursor, mysql_connection)
+    # insert data
+    insert_data(mysql_cursor)
+
+    # Commit transaction
+    mysql_connection.commit()
+
+    # Close cursor
+    mysql_cursor.close()
+    # Close connection
+    mysql_connection.close()
+
+    # Print completed
+    print("Database set up completed.")
+
+if __name__ == '__main__': main()
```

This allows exceptions to reach the console when true. You should have exception handling here.

---

Should I just catch all exceptions?

---

It would be better if you caught specific exceptions that you were expecting so you can print to console helpful output.

---

Got it